asserts/repair_test.go

```diff
@@ -0,0 +1,130 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/asserts"
+	. "gopkg.in/check.v1"
+)
+
+var (
+	_ = Suite(&repairSuite{})
+)
+
+type repairSuite struct {
+	modelsLine string
+
+	repairStr string
+}
+
+const script = `#!/bin/sh
+set -e
+echo "Unpack embedded payload"
+match=$(grep --text --line-number '^PAYLOAD:$' $0 | cut -d ':' -f 1)
+payload_start=$((match + 1))
+# Using "base64" as its part of coreutils which should be available
+# everywhere
+tail -n +$payload_start $0 | base64 --decode - | tar -xzf -
+# run embedded content
+./hello
+exit 0
+# payload generated with, may contain binary data
+#   printf '#!/bin/sh\necho hello from the inside\n' > hello
+#   chmod +x hello
+#   tar czf - hello | base64 -
+PAYLOAD:
+H4sIAJJt+FgAA+3STQrCMBDF8ax7ihEP0CkxyXn8iCZQE2jr/W11Iwi6KiL8f5u3mLd4i0mx76tZ
+l86Cc0t2welrPu2c6awGr95bG4x26rw1oivveriN034QMfFSy6fet/uf2m7aQy7tmJp4TFXS8g5y
+HupVphQllzGfYvPrkQAAAAAAAAAAAAAAAACAN3dTp9TNACgAAA==
```

Comment:
I just had an 80s flashback here.. with uuencode, usenet, ... Thanks for the test, and the mind trip! :)